src/platform/packages/private/kbn-controls-renderer/src/components/control_panel.tsx

```diff
@@ -174,7 +182,14 @@ export const ControlPanel = ({
             })}
             css={styles.formControl}
             prepend={
-              <>
+              <DisplaySettingsPopover
+                grow={grow}
+                width={width}
+                parentApi={parentApi}
+                uuid={uuid}
+                isOpen={isDisplaySettingsPopoverOpen}
+                closePopover={closeDisplaySettingsPopover}
+              >
```

Comment:
I don't love this. I think we should still try to use `MenuItem` in the action itself to render the popover without any wrapping elements. I **thought** that popovers had a way of setting the anchor via a reference but I'm not seeing that in the EUI docs anymore.... I wonder if they removed it? Might be worth digging into.

But on second thought, if this is the only way to get the anchor to the control itself, maybe it is better to do the alternative that I suggested [here](https://github.com/elastic/kibana/pull/240149#issuecomment-3433499438) after all? i.e. just move the gear icon to be anchored to the left of the control

Reply:
Depending on LOE here, I think it might actually be faster and easier to just move the gear icon to hover on the left - this would clean up this code significantly (no more wrapping popover and no need to figure out how to pass references to the action), **and** the popover could remain attached to the gear icon.

You should be able to request the settings action by `id` and hardcode its left positioning without making it generic (i.e. it can be built to only ever support this one action). We should have plenty of space even for `small` controls to do this. What do you think @Zacqary?